DOCSP-40348: disallow commas in oidc prop val (#553)
mongodb/docs-java: source/includes/fundamentals/oidc-value-comma.rst

.. note::

   The ``authMechanismProperties`` connection string option takes a list
   of key-value pairs. A *value* in any pair must not contain a comma

>    (``,``) character, to prevent the risk that the driver parses your

Reviewer: ```suggestion
   (``,``) character, to prevent the driver from parsing your
```

   connection string ambiguously. A value can contain a colon (``:``)
   character, however.

   For example, the following code shows a connection string that the

>    driver might parse incorrectly:

Reviewer 2: This note seems too large. Consider the existing text,

> Replace the <percent-encoded audience> placeholder with the percent-encoded value of the audience server parameter configured on your MongoDB deployment.

We could have a large note explaining which characters would need to be percent-encoded, and which are allowed, and an example of an invalid string. However, we communicate all of this by adding the adjective "percent-encoded" in two places inline with the normal text. The note is also a bit confusing, because the parsing is not ambiguous, the issue is that `,` is a reserved character whether or not it is encoded, and whether it is in the value or the key. Also, the note location is above both tabs, but it should only be inside the connection string tab, since it only applies there.

Let's say:

> Replace the <percent-encoded audience> placeholder with the percent-encoded value of the audience server parameter configured on your MongoDB deployment. The comma (`,`) and its encoding (`%2C`) are reserved: if the unencoded value contains a comma (`,`), it must not be specified in the connection string, but in a ``MongoCredential`` instance, as demonstrated in the :guilabel:`MongoCredential` tab.

Author: Thank you for clarifying!

   .. code-block:: java
      :emphasize-lines: 4

>       String uri =

Author: Will wait for technical review to address this question

>       String uri =

Reviewer 2: The value must always be percent-encoded. The existing docs correctly state "Replace the placeholder with the percent-encoded value of the audience server parameter". However, the behaviour is that commas will now be parsed as a new property, even when encoded.
